xcell/mappers/mapper_HSC_DR1wl.py

```diff
@@ -167,119 +154,102 @@ def _clean_raw_catalog(self, fnames):
             cats.append(c)
         return vstack(cats)
 
+    def _get_ellip_maps(self):
+        print(f'Computing bin {self.bn} signal map')
+        cat = self.get_catalog()
+        we1 = get_map_from_points(cat, self.nside,
+                                  w=cat['e1']*cat[self.w_name],
+                                  ra_name='ra',
+                                  dec_name='dec')
+        we2 = get_map_from_points(cat, self.nside,
+                                  w=cat['e2']*cat[self.w_name],
+                                  ra_name='ra',
+                                  dec_name='dec')
+        mask = self.get_mask()
+        goodpix = mask > 0
+        we1[goodpix] /= mask[goodpix]
+        we2[goodpix] /= mask[goodpix]
+        return we1, we2
+
     def get_signal_map(self):
         if self.signal_map is None:
-            fname_1 = f'HSCDR1wl_{self.bn}_e1_ns{self.nside}.fits.gz'
-            fname_2 = f'HSCDR1wl_{self.bn}_e2_ns{self.nside}.fits.gz'
-            read_lite1, fname_lite1 = self._check_lite_exists(fname_1)
-            read_lite2, fname_lite2 = self._check_lite_exists(fname_2)
-            if read_lite1 and read_lite2:
-                print(f'Loading lite maps {fname_1}, {fname_2}')
-                e1 = hp.read_map(fname_lite1)
-                e2 = hp.read_map(fname_lite2)
-                self.signal_map = [e1, e2]
-            else:
-                print(f'Computing bin {self.bn} signal map')
-                cat = self.get_catalog()
-                we1 = get_map_from_points(cat, self.nside,
-                                          w=cat['e1']*cat[self.w_name],
-                                          ra_name='ra',
-                                          dec_name='dec')
-                we2 = get_map_from_points(cat, self.nside,
-                                          w=cat['e2']*cat[self.w_name],
-                                          ra_name='ra',
-                                          dec_name='dec')
-                mask = self.get_mask()
-                goodpix = mask > 0
-                we1[goodpix] /= mask[goodpix]
-                we2[goodpix] /= mask[goodpix]
-                self.signal_map = [we1, we2]
-                if fname_lite1 is not None:
-                    hp.write_map(fname_lite1, we1, overwrite=True)
-                if fname_lite2 is not None:
-                    hp.write_map(fname_lite2, we2, overwrite=True)
-
+            fn = f'HSCDR1wl_signal_{self.bn}_ns{self.nside}.fits.gz'
+            d = self._rerun_read_cycle(fn, 'FITSMap',
+                                       self._get_ellip_maps,
+                                       section=[0, 1])
+            self.signal_map = [d[0], d[1]]
```

Comment:
I'm fine with it being an np.array, but not a tuple. Otherwise namaster won't like it and we'll need to add extra code to transform into a list. I've opened an issue so we do it later